### DISCUSSION TITLE
State of Bunch toggle visible in Stream Deck?

### DISCUSSION THREAD

**Discussion by kontiainen** [ttscoff/bunch-docs#69] [Q&A]
Would love to understand how to (easily) get Bunch toggle state visible on a Stream Deck button. 

I've found Stream Deck [SDK documentation](https://developer.elgato.com/documentation/stream-deck/sdk/events-sent/#settitle), and setTitle and setImage stuff from there, but as a newbie programmer the bar is quite high... I'd appreciate a complete example instead - or would love to hear if there's some other approach.

**Comment by ttscoff** [comment]
Bunch doesn't currently offer a way to retrieve the state of a Bunch. It would be possible to add this to the AppleScript dictionary, I think, but right now it's mostly one-way (you can send commands, but not get states other than a full list of available Bunches). If it were available, you could potentially use Keyboard Maestro to change the state of the image, and I have seen Stream Deck plugins that use AppleScript themselves, though I've never played with it. But I will consider adding a `get bunch` command that provides state and whatever else I want to expose.

That said, Bunch does expose state internally for its [conditional logic](https://bunchapp.co/docs/bunch-files/logic/), and Keyboard Maestro can update Stream Deck buttons. So... if you had a non-toggling Bunch that launched at repeated intervals, it could use KM to update the buttons based on conditions:

```
---
title: Stream Deck Updater
toggles: false
open every: 1m
---
if Comms is open
    * [applescript or url handler to trigger a KM macro that sets the Comms button image to on]
else
    * [applescript or url handler to trigger a KM macro that sets the Comms button image to off]
end

if ... is open...
```

Not tested, but I would think it's a feasible option.
